src/main/java/sg/ncl/MainController.java

```diff
@@ -2105,113 +2102,123 @@ public String data(Model model, HttpSession session) throws Exception {
-//			} finally {
-//                if (stream != null) {
-//                    stream.close();
-//                }
-//                if (fileOutputStream != null) {
-//                    fileOutputStream.close();
-//                }
-//            }
-//        }
-//		else {
-//			redirectAttributes.addFlashAttribute("message",
-//					"You failed to upload " + file.getOriginalFilename() + " because the file was empty");
-//		}
-//    	return "redirect:/data";
-//    }
+        ResponseEntity response = getResponseEntity(id, request);
+        String dataResponseBody = response.getBody().toString();
 
-//    @RequestMapping(value="/data/edit/{datasetId}", method=RequestMethod.GET)
-//    public String datasetInfo(@PathVariable Integer datasetId, Model model) {
-//    	Dataset dataset = datasetManager.getDataset(datasetId);
-//    	model.addAttribute("editDataset", dataset);
-//    	return "edit_data";
-//    }
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
 
-//    @RequestMapping(value="/data/edit/{datasetId}", method=RequestMethod.POST)
-//    public String editDatasetInfo(@PathVariable Integer datasetId, @ModelAttribute("editDataset") Dataset dataset, final RedirectAttributes redirectAttributes) {
-//    	Dataset origDataset = datasetManager.getDataset(datasetId);
-//
-//    	String editedDatasetName = dataset.getDatasetName();
-//    	String editedDatasetDesc = dataset.getDatasetDescription();
-//    	String editedDatasetLicense = dataset.getLicense();
-//    	String editedDatasetPublic = dataset.getIsPublic();
-//    	boolean editedDatasetIsRequiredAuthorization = dataset.getRequireAuthorization();
-//
-//    	System.out.println(origDataset.getDatasetId());
-//    	System.out.println(dataset.getDatasetId());
-//
-//    	if (origDataset.updateName(editedDatasetName) == true) {
-//    		redirectAttributes.addFlashAttribute("editName", "success");
-//    	}
-//
-//    	if (origDataset.updateDescription(editedDatasetDesc) == true) {
-//    		redirectAttributes.addFlashAttribute("editDesc", "success");
-//    	}
-//
-//    	if (origDataset.updateLicense(editedDatasetLicense) == true) {
-//    		redirectAttributes.addFlashAttribute("editLicense", "success");
-//    	}
-//
-//    	if (origDataset.updatePublic(editedDatasetPublic) == true) {
-//    		redirectAttributes.addFlashAttribute("editPublic", "success");
-//    	}
-//
-//    	if (origDataset.updateAuthorization(editedDatasetIsRequiredAuthorization) == true) {
-//    		redirectAttributes.addFlashAttribute("editIsRequiredAuthorization", "success");
-//    	}
-//
-//    	datasetManager.updateDatasetDetails(origDataset);
-//
-//    	return "redirect:/data/edit/{datasetId}";
-//    }
+                switch (exceptionState) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1301)